com.sap.cloud.lm.sl.cf.persistence/src/main/java/com/sap/cloud/lm/sl/cf/persistence/services/ProcessLogger.java

```diff
@@ -0,0 +1,83 @@
+package com.sap.cloud.lm.sl.cf.persistence.services;
+
+import java.io.File;
+
+import org.apache.commons.io.FileUtils;
+import org.apache.log4j.Logger;
+import org.flowable.engine.delegate.DelegateExecution;
+
+import com.sap.cloud.lm.sl.cf.persistence.message.Constants;
+
+public class ProcessLogger extends Logger {
+
+    private Logger logger;
+    private File log;
+    private String logName;
+    private String processId;
+    private String activityId;
+
+    public ProcessLogger(Logger logger, File log, String logName, String processId, String activityId) {
+        super(logger.getName());
+        this.logger = logger;
+        this.log = log;
+        this.logName = logName;
+        this.processId = processId;
+        this.activityId = activityId;
+    }
+
+    @Override
```

Comment:
A decorator-like ProcessLogger in order to easily track which logger is for which process/activity, and which file it logs to. Additionally added functionalities for deletion of file etc..
The methods need to be overriden because of the way Logger is created (Logger.getLogger), adding the appender, etc.. Overriden in order to use the originally created logger.